File: src/components/JSONHandler/src/AddSubMenu.cpp
#include "JSONHandler/AddSubMenu.h"
#include "JSONHandler/RPC2Marshaller.h"

using namespace RPC2Communication;

AddSubMenu::AddSubMenu()
:RPC2Request( RPC2Marshaller :: METHOD_ADDSUBMENU_REQUEST )
{
    position = 0;
}
 
AddSubMenu::~AddSubMenu()
{
    if(position)
    {
        delete position;
    }
}

int AddSubMenu::getMenuId() const
{
    return menuId;
}

int * AddSubMenu::getPosition() const
{
    return position;
}

const std::string &  AddSubMenu::getMenuName() const
{
    return menuName;
}

void AddSubMenu::setMenuId( int menuId )
{
    this->menuId = menuId;
}

void AddSubMenu::setPosition( int  position )
{
    if(this->position)
    {
        delete this->position;
    }
        
    this->position = new int(position);
}

void AddSubMenu::setMenuName( const std::string & menuName )
{
    this->menuName = menuName;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/AddSubMenu.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCRequest;
import com.ford.syncV4.proxy.constants.Names;

public class AddSubMenu extends RPCRequest {

    public AddSubMenu() {
        super("AddSubMenu");
    }
    public AddSubMenu(Hashtable hash) {
        super(hash);
    }
    public Integer getMenuID() {
        return (Integer) parameters.get( Names.menuID );
    }
    public void setMenuID( Integer menuID ) {
        if (menuID != null) {
            parameters.put(Names.menuID, menuID );
        }
    }
    public Integer getPosition() {
        return (Integer) parameters.get( Names.position );
    }
    public void setPosition( Integer position ) {
        if (position != null) {
            parameters.put(Names.position, position );
        }
    }
    public String getMenuName() {
        return (String) parameters.get( Names.menuName );
    }
    public void setMenuName( String menuName ) {
        if (menuName != null) {
            parameters.put(Names.menuName, menuName );
        }
    }
}
